main.go

```diff
@@ -58,10 +59,31 @@ func (lf logFunc) Println(v ...interface{}) {
 	lf("msg", fmt.Sprintln(v...))
 }
 
+func decompressGzip() func(http.Handler) http.Handler {
+	return func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			contentEncoding := r.Header.Get("Content-Encoding")
+			if strings.Contains(contentEncoding, "gzip") {
+				gr, err := gzip.NewReader(r.Body)
+				if err != nil {
+					w.WriteHeader(http.StatusBadRequest)
+					w.Write([]byte(err.Error()))
+				} else {
+					r.Body = gr
+					h.ServeHTTP(w, r)
+				}
+			} else {
+				h.ServeHTTP(w, r)
+			}
+		})
+	}
+}
+
 func main() {
 	var (
 		app                 = kingpin.New(filepath.Base(os.Args[0]), "The Pushgateway")
 		webConfig           = webflag.AddFlags(app)
+		acceptGzip          = app.Flag("web.accept-gzip", "Web server can handle gzip content or not").Default("false").Bool() // gzip.DefaultCompression
```

Comment:
As mentioned above, I think we don't need a flag after all.

Reply:
ok